app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -239,33 +238,21 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
     }
 
     override fun navToQuickSettingsSheet(tab: SessionState, sitePermissions: SitePermissions?) {
-        val directions =
-            BrowserFragmentDirections.actionBrowserFragmentToQuickSettingsSheetDialogFragment(
-                sessionId = tab.id,
-                url = tab.content.url,
-                title = tab.content.title,
-                isSecured = tab.content.securityInfo.secure,
-                sitePermissions = sitePermissions,
-                gravity = getAppropriateLayoutGravity(),
-                certificateName = tab.content.securityInfo.issuer,
-                permissionHighlights = tab.content.permissionHighlights
-            )
-        nav(R.id.browserFragment, directions)
-    }
-
-    override fun navToTrackingProtectionPanel(tab: SessionState) {
-        val navController = findNavController()
-
         requireComponents.useCases.trackingProtectionUseCases.containsException(tab.id) { contains ->
-            val isEnabled = tab.trackingProtection.enabled && !contains
+            val isTrackingProtectionEnabled = tab.trackingProtection.enabled && !contains
             val directions =
-                BrowserFragmentDirections.actionBrowserFragmentToTrackingProtectionPanelDialogFragment(
+                BrowserFragmentDirections.actionBrowserFragmentToQuickSettingsSheetDialogFragment(
                     sessionId = tab.id,
                     url = tab.content.url,
-                    trackingProtectionEnabled = isEnabled,
-                    gravity = getAppropriateLayoutGravity()
+                    title = tab.content.title,
+                    isSecured = tab.content.securityInfo.secure,
+                    sitePermissions = sitePermissions,
+                    gravity = getAppropriateLayoutGravity(),
+                    certificateName = tab.content.securityInfo.issuer,
+                    permissionHighlights = tab.content.permissionHighlights,
+                    isTrackingProtectionEnabled = isTrackingProtectionEnabled
                 )
-            navController.navigateSafe(R.id.browserFragment, directions)
+            nav(R.id.browserFragment, directions)
```

Comment:
Is there any particular reason why we are replacing `navigateSafe()` by `nav()`.

Reply:
Oh, I can see why the diff looks weird. I actually deleted `navToTrackingProtectionPanel` and was working in `navToQuickSettingsSheet` which uses `nav`. Otherwise, `navigateSafe` really does the same thing as `nav` if you look into much more deeper. We really should clean this all up.

Reply:
Thanks for the clarification, maybe we can open a ticket to clear this later

Reply:
Filed https://github.com/mozilla-mobile/fenix/issues/20039